attestation/src/certificate.rs

```diff
@@ -130,7 +130,7 @@ impl<'a> Certificate<'a> {
         cdi_id: &'a [u8],
         key_pair: &'a [u8],
         certificate_buf: &'a mut [u8],
-    ) -> Result<'a, &'a [u8]> {
+    ) -> Result<&'a [u8]> {
```

Comment:
Nit: Consider rephrasing the commit message "That allows for removing the lifetime bound from the crate Error
enum." as "This eliminates the lifetime bound from...".